messagix/responsehandler.go

```diff
@@ -19,6 +20,10 @@ type ResponseHandler struct {
 	packetChannels  map[uint16]chan interface{}
 }
 
+var ErrTimeout = fmt.Errorf("timeout waiting for response")
+var ErrContextCancelled = fmt.Errorf("context cancelled before response received")
+var ErrChannelNotFound = fmt.Errorf("channel not found for packet")
```

Comment:
Happy to take any suggestions for better error messages here.

Reply:
Could say "client disconnected" or something instead of context cancelled. The two others are probably fine